flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/exceptionhistory/RootExceptionHistoryEntry.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.exceptionhistory;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * {@code RootExceptionHistoryEntry} extending {@link ExceptionHistoryEntry} by providing a list of
+ * {@code ExceptionHistoryEntry} instances to store concurrently caught failures.
+ */
+public class RootExceptionHistoryEntry extends ExceptionHistoryEntry {
+
+    private static final long serialVersionUID = -7647332765867297434L;
+
+    private final Collection<ExceptionHistoryEntry> concurrentExceptions = new ArrayList<>();
+
+    /**
+     * Creates a {@code ExceptionHistoryEntry} representing a global failure from the passed {@code
+     * Throwable} and timestamp.
+     *
+     * @param cause The reason for the failure.
+     * @param timestamp The time the failure was caught.
+     * @return The {@code ExceptionHistoryEntry} instance.
+     */
+    @VisibleForTesting
+    public static RootExceptionHistoryEntry fromGlobalFailure(Throwable cause, long timestamp) {
+        return new RootExceptionHistoryEntry(cause, timestamp, null, null);
+    }
+
+    @VisibleForTesting
+    public RootExceptionHistoryEntry(
+            Throwable cause,
+            long timestamp,
+            @Nullable String failingTaskName,
+            @Nullable TaskManagerLocation taskManagerLocation) {
+        super(cause, timestamp, failingTaskName, taskManagerLocation);
+    }
+
+    public void add(ExceptionHistoryEntry concurrentException) {
```

Comment:
```suggestion
    void add(ExceptionHistoryEntry concurrentException) {
```
Ideally we get rid of it though.